### PR TITLE
Implemented Simulated CACU

### PR DESCRIPTION
**Pull request by marvin-steinke**
This pull request introduces a new *simulated* carbon-aware control unit (CACU) and enhances the functionality of the mock power meter in the cosimulation example.

Details:

A new carbon-aware control unit (CACU) is introduced. The CACU simulator and its associated model control the power mode of the mock power meters and the state of charge (SOC) of the storage based on the carbon intensity and current time.

The `MockPowerMeter` class is enhanced to support different power modes that are: 'high performance', 'normal', and 'power-saving'. Each power mode corresponds to a different power consumption factor. An additional method `set_power_mode` has been added to allow changing the power mode.

Refactoring includes renaming `carbon_aware_control_unit.py` to `cacu.py` for simplicity.

### CODE REVIEW

**Comment by marvin-steinke** [comment]
checks fail because of #115

**Review by amandamalk0601 — APPROVED** [review]

**Comment by marvin-steinke** [comment]
#118

File: vessim/core/consumer.py [hunk 1]
```diff
@@ -62,7 +120,7 @@ class ComputingSystem(Consumer):
         pue: The power usage effectiveness of the system.
     """
 
-    def __init__(self, power_meters: List[PowerMeter], pue: float = 1):
+    def __init__(self, power_meters: list[PowerMeter], pue: float = 1):
```
Comment (birnbaum) [inline vessim/core/consumer.py:123]: this is Python 3.9 code, but we also want to support 3.8

File: vessim/cosim/cacu.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        else:
+            self.storage.min_soc = .6
+
+        for power_meter in self.mock_power_meters:
+            if inputs["ci"] <= 200 or self.storage.soc() > .8:
+                power_meter.set_power_mode("high performance")
+            elif inputs["ci"] >= 250 and self.storage.soc() < self.storage.min_soc:
+                power_meter.set_power_mode("normal")
+            else:
+                power_meter.set_power_mode("power-saving")
```
Comment (birnbaum) [inline vessim/cosim/cacu.py:64]: is this logic now duplicated between the simulated and SiL scenarios?

**Review by birnbaum — CHANGES_REQUESTED** [review]

File: vessim/cosim/cacu.py (hunk 2 above)
Comment (marvin-steinke) [reply vessim/cosim/cacu.py:64]: Basically yes. We could build an interface class with a configurable scenario use case that is then used by the simulated and sil cacu. However that class wouldn't be as trivial and I suggest moving that task to a seperate PR. I'll open an issue.

File: vessim/core/consumer.py (hunk 1 above)
Comment (marvin-steinke) [reply vessim/core/consumer.py:132]: Yes sry, I really have to configure my linter for that.

File: vessim/core/consumer.py [hunk 3]
```diff
@@ -27,18 +27,76 @@ def finalize(self) -> None:
 
 
 class MockPowerMeter(PowerMeter):
+    """A mock power meter class.
```
[the remark this answers is not in the document]
Comment (marvin-steinke) [reply vessim/core/consumer.py]: What dir do you propose, it doesn't quite fit into exising ones. maybe ```examples/core/```?

**Review by birnbaum — APPROVED** [review]

**Pull request merged**